File: vertx-elasticsearch/src/main/java/fr/myprysm/vertx/elasticsearch/action/get/GetRequestItem.java
package fr.myprysm.vertx.elasticsearch.action.get;


import fr.myprysm.vertx.elasticsearch.action.support.FetchSourceContext;
import io.vertx.codegen.annotations.DataObject;
import io.vertx.core.json.JsonObject;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import lombok.experimental.Accessors;
import org.elasticsearch.common.lucene.uid.Versions;
import org.elasticsearch.index.VersionType;

import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Accessors(chain = true)
@EqualsAndHashCode
@DataObject(generateConverter = true)
public class GetRequestItem {

    private String index;
    private String type;
    private String id;
    private String routing;
    private String parent;
    private List<String> storedFields;
    private long version = Versions.MATCH_ANY;
    private VersionType versionType = VersionType.INTERNAL;
    private FetchSourceContext fetchSourceContext;

    public GetRequestItem(String index, String type, String id) {
        this.index = index;
        this.type = type;
        this.id = id;
    }

    public GetRequestItem(JsonObject json) {
        GetRequestItemConverter.fromJson(json, this);
    }

    public JsonObject toJson() {
        JsonObject json = new JsonObject();
        GetRequestItemConverter.toJson(this, json);
        return json;
    }
}

File: vertx-elasticsearch/src/main/kotlin/fr/myprysm/vertx/elasticsearch/kotlin/action/get/GetRequestItem.kt
package fr.myprysm.vertx.elasticsearch.kotlin.action.get

import fr.myprysm.vertx.elasticsearch.action.get.GetRequestItem
import fr.myprysm.vertx.elasticsearch.action.support.FetchSourceContext
import org.elasticsearch.index.VersionType

fun GetRequestItem(
  fetchSourceContext: fr.myprysm.vertx.elasticsearch.action.support.FetchSourceContext? = null,
  id: String? = null,
  index: String? = null,
  parent: String? = null,
  routing: String? = null,
  storedFields: Iterable<String>? = null,
  type: String? = null,
  version: Long? = null,
  versionType: VersionType? = null): GetRequestItem = fr.myprysm.vertx.elasticsearch.action.get.GetRequestItem().apply {

  if (fetchSourceContext != null) {
    this.setFetchSourceContext(fetchSourceContext)
  }
  if (id != null) {
    this.setId(id)
  }
  if (index != null) {
    this.setIndex(index)
  }
  if (parent != null) {
    this.setParent(parent)
  }
  if (routing != null) {
    this.setRouting(routing)
  }
  if (storedFields != null) {
    this.setStoredFields(storedFields.toList())
  }
  if (type != null) {
    this.setType(type)
  }
  if (version != null) {
    this.setVersion(version)
  }
  if (versionType != null) {
    this.setVersionType(versionType)
  }
}

